File: linear_search.java
public class LinearSearch {

    /**
     * Performs a linear search on an array to find a target element.
     *
     * @param arr The array to search within.
     * @param target The element to search for.
     * @return The index of the target element if found, otherwise -1.
     */
    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i; // Element found, return its index
            }
        }
        return -1; // Element not found
    }

    public static void main(String[] args) {
        int[] numbers = {10, 25, 8, 42, 15, 7};
        int target1 = 42;
        int target2 = 99;

        int index1 = linearSearch(numbers, target1);
        if (index1 != -1) {
            System.out.println(target1 + " found at index: " + index1);
        } else {
            System.out.println(target1 + " not found in the array.");
        }

        int index2 = linearSearch(numbers, target2);
        if (index2 != -1) {
            System.out.println(target2 + " found at index: " + index2);
        } else {
            System.out.println(target2 + " not found in the array.");
        }
    }
}

File: linear_search.c
#include <stdio.h>

// Function to perform linear search
int linearSearch(int arr[], int n, int key) {
    // Iterate through the array from the first element to the last
    for (int i = 0; i < n; i++) {
        // If the key is found, return its index
        if (arr[i] == key) {
            return i;
        }
    }
    // If the key is not found after traversing the entire array, return -1
    return -1;
}

int main() {
    int arr[] = {10, 20, 80, 30, 60, 50, 110, 100, 130, 170};
    int n = sizeof(arr) / sizeof(arr[0]); // Calculate the size of the array
    int key = 110; // The element to search for

    // Call the linearSearch function
    int result = linearSearch(arr, n, key);

    // Print the result based on the value returned by the function
    if (result == -1) {
        printf("Element %d is not present in the array.\n", key);
    } else {
        printf("Element %d found at index: %d\n", key, result);
    }

    // Example for an element not found
    key = 99;
    result = linearSearch(arr, n, key);
    if (result == -1) {
        printf("Element %d is not present in the array.\n", key);
    } else {
        printf("Element %d found at index: %d\n", key, result);
    }

    return 0;
}
